plutus-conformance/agda/Spec.hs

```diff
@@ -3,91 +3,92 @@
 {- | Conformance tests for the Agda implementation. -}
 module Main (main) where
 
-import Control.Monad.Trans.Except
+import Control.Monad.Trans.Except (ExceptT, runExceptT, withExceptT)
+
 import MAlonzo.Code.Evaluator.Term (runUCountingAgda)
-import PlutusConformance.Common
+-- import MAlonzo.Code.Cost.Raw (HRawCostModel)
+
+import PlutusConformance.Common (UplcEvaluator (..), runUplcEvalTests)
 import PlutusCore (Error (..))
-import PlutusCore.Default
-import PlutusCore.Evaluation.Machine.CostModelInterface
-import PlutusCore.Evaluation.Machine.ExBudget
-import PlutusCore.Evaluation.Machine.ExBudgetingDefaults (defaultCekMachineCosts)
-import PlutusCore.Evaluation.Machine.ExMemory
-import PlutusCore.Evaluation.Machine.SimpleBuiltinCostModel
+import PlutusCore.Default (DefaultFun, DefaultUni)
+import PlutusCore.Evaluation.Machine.CostModelInterface (CekMachineCosts, CostModelParams,
+                                                         applyCostModelParams)
+import PlutusCore.Evaluation.Machine.ExBudget (ExBudget (..))
+import PlutusCore.Evaluation.Machine.ExBudgetingDefaults (defaultCekCostModel)
+import PlutusCore.Evaluation.Machine.ExMemory (ExCPU (..), ExMemory (..))
+import PlutusCore.Evaluation.Machine.MachineParameters (CostModel (..))
+import PlutusCore.Evaluation.Machine.SimpleBuiltinCostModel (BuiltinCostKeyMap, BuiltinCostMap,
+                                                             toSimpleBuiltinCostModel)
 
 import PlutusCore.Quote
 import UntypedPlutusCore qualified as UPLC
 import UntypedPlutusCore.DeBruijn
 
--- -- | Our `evaluator` for the Agda UPLC tests is the CEK machine without costs.
--- agdaEvalUplcProgNoCost :: UplcEvaluator
--- agdaEvalUplcProgNoCost = UplcEvaluatorWithoutCosting $ \(UPLC.Program () version tmU) ->
---     let
---         -- turn it into an untyped de Bruijn term
---         tmUDB :: ExceptT FreeVariableError Quote (UPLC.Term NamedDeBruijn DefaultUni DefaultFun ())
---         tmUDB = deBruijnTerm tmU
---     in
---     case runQuote $ runExceptT $ withExceptT FreeVariableErrorE tmUDB of
---         -- if there's an exception, evaluation failed, should return `Nothing`.
---         Left _ -> Nothing
---         -- evaluate the untyped term with CEK
---         Right tmUDBSuccess ->
---             case runUAgda tmUDBSuccess of
---                 Left _ -> Nothing
---                 Right tmEvaluated ->
---                     let tmNamed = runQuote $ runExceptT $
---                             withExceptT FreeVariableErrorE $ unDeBruijnTerm tmEvaluated
---                     in
---                     -- turn it back into a named term
---                     case tmNamed of
---                         Left _          -> Nothing
---                         Right namedTerm -> Just $ UPLC.Program () version namedTerm
```

Comment:
I had left the evaluator without costing just in case in the future we need to run momentarily the conformance test without costing (for example because the cost model in the metatheory is not in synch with the Haskell implementation).

Reply:
Oh, I didn't notice the difference!  Now that you've mentioned it, it's tempting to try and make a single function that does both, but on closer examination that's maybe not so easy.

Reply:
@mjaskelioff I've put the evaluator without costing back and added a switch to let you turn it on and off easily.